### PR TITLE
Extract zip files to cache folder

### LINKED ISSUE #400: Extract zip files to cache rather than `/tmp`
Yeah, I was a bit worried about the clobbering thing, which is why I wanted to use a unique folder for each extraction.

I'm planning on adding caching for downloaded zip files at some point, so could also use the cache folder for extracing them to? On Linux this would be something like `.cache/healthgps`. We'd have to be a little careful to make sure there aren't race conditions with extracting and reading the zip files (e.g. for array jobs), but this should be doable. What do you think?

_Originally posted by @alexdewar in https://github.com/imperialCHEPI/healthgps/pull/391#discussion_r1620537989_

### PR DESCRIPTION
**Pull request by alexdewar**
By extracting zip files to a cache folder (e.g. on Linux, `.cache/healthgps`), we can avoid having to extract the same file more than once. While the performance benefits of this are negligible, if they exist at all, I'm introducing it as a preparatory step for when we want to be able to download data sets from a URL, with a given hash. (In this case the download can be avoided if the cache folder already exists.) For the hashing, I'm using SHA256 hashes, generated using openssl, which is already an indirect dependency.

For generating a platform-appropriate cache directory, I used the PlatformFolders library.

Closes #400.

### CODE REVIEW

**Comment by alexdewar** [comment]
Hmm... one of the tests is failing on Windows only. How odd. I'll dig into it when I get a chance

**Review by jamesturner246 — APPROVED** [review]
LGTM :+1: 

Nice -- very elegant. Just a question.

File: src/HealthGPS.Tests/Sha256.Test.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+TEST(SHA256, SHA256Context) {
+    // Empty hash
+    {
+        hgps::SHA256Context ctx;
+        EXPECT_EQ(ctx.finalise(),
+                  "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855");
+    }
+
+    // Example string
+    {
```
[the remark this answers is not in the document]
Comment (alexdewar) [reply src/HealthGPS.Tests/Sha256.Test.cpp:16]: It's exactly that. And if you try removing the line ending, `pre-commit` will re-add it :stuck_out_tongue:

[the remark this answers is not in the document]
Comment (jamesturner246) [reply src/HealthGPS.Tests/Sha256.Test.cpp:16]: Oh! xD

**Comment by jamesturner246** [comment]
I know it's not super pretty, but how about moving all the testsuite data files into a `src/HealthGPS.Tests/test_files` directory, and using a pre-commit exclude filter on that directory?

**Pull request merged**